### PR TITLE
tools: add the doc of DM-worker intro

### PR DESCRIPTION
**Pull request by CaitinChen**
Via https://github.com/pingcap/tidb-tools/pull/129
@lilin90 @GregoryIan PTAL

### CODE REVIEW

**Comment by GregoryIan** [comment]
@kennytm PTAL

File: tools/dm-worker-intro.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+---
+title: DM-worker Introduction
+summary: Learn the features of DM-worker.
+category: tools
+---
+
+# DM-worker Introduction
+
+DM-worker is a tool used to synchronize the data from MySQL/MariaDB in the upstream to TiDB in the downstream.
```
Comment (GregoryIan) [inline tools/dm-worker-intro.md]: is `from MySQL/MariaDB to TiDB` enough?

File: tools/dm-worker-intro.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+---
+
+# DM-worker Introduction
+
+DM-worker is a tool used to synchronize the data from MySQL/MariaDB in the upstream to TiDB in the downstream. 
+
+It has the following features:
+
+- Connecting to a MySQL/MariaDB instance and registering as the slave of this instance
+- Reading the binlog files of MySQL/MariaDB and persisting the binlog files to the local storage (relay logs)
```
Comment (GregoryIan) [inline tools/dm-worker-intro.md]: `Reading the binlog events from MySQL/MariaDB and persisting them to the local storage`

File: tools/dm-worker-intro.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+- A single DM-worker supports synchronizing the data of one MySQL/MariaDB instance in the upstream to multiple TiDB instances in the downstream
```
Comment (GregoryIan) [inline tools/dm-worker-intro.md]: ditto, is `from xx to TiDB` enough?

File: tools/dm-worker-intro.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+category: tools
+---
+
+# DM-worker Introduction
+
+DM-worker is a tool used to synchronize the data from MySQL/MariaDB in the upstream to TiDB in the downstream. 
+
+It has the following features:
+
+- Connecting to a MySQL/MariaDB instance and registering as the slave of this instance
```
Comment (kennytm) [inline tools/dm-worker-intro.md]: Present tense? 🙃

```suggestion
- Connects to a MySQL/MariaDB instance and registers as its slave
```

**Comment by CaitinChen** [comment]
@kennytm @GregoryIan @morgo Thanks!

File: tools/dm-worker-intro.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+```sql
+GRANT SELECT,INSERT,UPDATE,DELETE,CREATE,DROP,ALTER,INDEX  ON *.* TO 'your_user'@'your_wildcard_of_host';
+```
+
+### Minimal privilege each processing unit needs
+
+| Processing Unit | Minimal Privilege for Upstream (MySQL/MariaDB) | Minimal Privilege for Downstream (TiDB) | Minimal Privilege for the System |
+|----:|:--------------------|:------------|:----|
+| Relay log | `SELECT` (checks some upstream environment variables, like `binlog_format`)<br>`REPLICATION SLAVE` (reads the binlog)<br>`REPLICATION CLIENT` (showa the master status and slave status) | NULL | Read/Write local files |
```
Comment (amyangfei) [inline tools/dm-worker-intro.md]: s/showa/shows

File: tools/dm-worker-intro.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+Its rationale and features are similar to the slave relay log of MySQL. For details, see [The Slave Relay Log](https://dev.mysql.com/doc/refman/5.7/en/slave-logs-relaylog.html).
+
+### Dumper
+
+Dumper dumps the full data from the upstream MySQL/MariaDB to the local disk.
+
+### Loader
+
+Loader reads the files of Dumper and then load these files to the downstream TiDB.
```
Comment (amyangfei) [inline tools/dm-worker-intro.md]: s/load/loads

File: tools/dm-worker-intro.md (hunk 5 above)
Comment (amyangfei) [reply tools/dm-worker-intro.md]: In fact these are two MySQL commands: `show master status` and `show slave status`
s/ showa the master status and slave status/show master status, show slave status

Comment (amyangfei) [reply tools/dm-worker-intro.md]: Besides, in this table, we can add `` to MySQL raw command, when it is in a (), in order to distinguish SQL command from descriptive words

**Comment by CaitinChen** [comment]
@amyangfei Thanks @~@

**Pull request merged**